flytekit/image_spec/image_spec.py

```diff
@@ -130,7 +130,7 @@ def tag(self) -> str:
         # copy the image spec to avoid modifying the original image spec. otherwise, the hash will be different.
         spec = copy.deepcopy(self)
         if isinstance(spec.base_image, ImageSpec):
-            spec = dataclasses.replace(spec, base_image=spec.base_image)
+            spec = dataclasses.replace(spec, base_image=spec.base_image.image_name())
```

Comment:
we used image_name before. I accidentally removed it in this [PR](https://github.com/flyteorg/flytekit/pull/2593/files#diff-4f7ad8410fca25108732e8275def1390a415572e69f59ff19d13c86c6af86106L368-L370)